cfgmgr/intfmgr.cpp

```diff
@@ -74,6 +72,70 @@ void IntfMgr::setIntfVrf(const string &alias, const string vrfName)
     EXEC_WITH_ERROR_THROW(cmd.str(), res);
 }
 
+void IntfMgr::addHostSubIntf(const string&intf, const string &subIntf, const string &vlan)
+{
+    stringstream cmd;
+    string res;
+
+    cmd << IP_CMD << " link add link " << intf << " name " << subIntf << " type vlan id " << vlan;
```

Comment:
>intf [](start = 42, length = 4)

Please validate user input before concat a command line

Reply:
I have fixed in [https://github.com/Azure/sonic-swss/pull/1098](https://github.com/Azure/sonic-swss/pull/1098). You may follow and simplify your code. This is optional

---
In reply to: [328377564](https://github.com/Azure/sonic-swss/pull/969#discussion_r328377564) [](ancestors = 328377564)